systemtest/src/test/java/io/strimzi/systemtest/CruiseControlST.java

```diff
@@ -4,28 +4,254 @@
+            "/bin/bash", "-c", ccStatusCommand).out();
+
+        LOGGER.info("Verifying that {} is running inside the Pod {} and REST API is available", CRUISE_CONTROL_NAME, ccPodName);
+
+        assertThat(result, not(containsString("404")));
+        assertThat(result, containsString("RUNNING"));
+
+        verifyThatCruiseControlTopicsArePresent();
+    }
+
+    @Test
+    void testUninstallingAndInstallationCruiseControl() {
+
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, kafka -> {
+            LOGGER.info("Removing Cruise Control to the classic Kafka.");
+            kafka.getSpec().setCruiseControl(null);
+        });
+
+        StatefulSetUtils.waitTillSsHasRolled(kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaPods);
+
+        LOGGER.info("Verifying that in {} is not present in the Kafka cluster", CRUISE_CONTROL_NAME);
+        assertThat(KafkaResource.kafkaClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getSpec().getCruiseControl(), nullValue());
+
+        LOGGER.info("Verifying that {} pod is not present", CRUISE_CONTROL_NAME);
+        assertThat(kubeClient().listPodsByPrefixInName(CRUISE_CONTROL_POD_PREFIX).size(), is(0));
+
+        // TODO: should topic be deleted ? CURRENTLY CC let the topics in the kafka
+//        verifyThatCruiseControlTopicsAreNotPresent();
+
+        kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, kafka -> {
+            LOGGER.info("Adding Cruise Control to the classic Kafka.");
+            kafka.getSpec().setCruiseControl(new CruiseControlSpec());
+        });
+
+        StatefulSetUtils.waitTillSsHasRolled(kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaPods);
+
+        LOGGER.info("Verifying that {} topics are created after CC is instantiated.", CRUISE_CONTROL_NAME);
+
+        KafkaTopicUtils.waitForKafkaTopicReady(CRUISE_CONTROL_METRICS_TOPIC);
+        KafkaTopicUtils.waitForKafkaTopicReady(CRUISE_CONTROL_MODEL_TRAINING_SAMPLES_TOPIC);
+        KafkaTopicUtils.waitForKafkaTopicReady(CRUISE_CONTROL_PARTITION_METRICS_SAMPLES_TOPIC);
+
+        verifyThatCruiseControlTopicsArePresent();
+    }
+
+    @Test
+    void testConfigurationChangeTriggersRollingUpdate() {
+
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, kafka -> {
+            LOGGER.info("Adding Cruise Control to the classic Kafka.");
+            BrokerCapacity brokerCapacity = kafka.getSpec().getCruiseControl().getBrokerCapacity();
+
+            // TODO: this doesn't work because in the Kafka CR there is nothing using `kafka.getSpec().getCruiseControl()` only object,
+            // TODO: which has some default values by they are not propagated to other fields...
```

Comment:
How is this handled for Kafka? Obviously, we don't want to update the CR with all the default values. CC prints all configs to its log at startup so you could `head` the log and `grep` for the setting?

Reply:
@see-quick can you elaborate a little bit more I am not sure I got what you mean.

Reply:
So the problem here is that whenever I want to customize the settings of the cruise control, for instance, change the broker capacity I am unable to do it because I will get the NPE. Is it desired? Because how can I change the properties of the Cruise Control?

Reply:
Can't we create a CruiseControlSpec and BrokerCapacity objects using the constructors? e.g. 
```
CruiseControlSpec cruiseControl = new CruiseControlSpec();
BrokerCapacity capacity = new BrokerCapacity();
capacity.setDisk("20000M");
cruiseControl.setCapacity(capacity);
kafka.getSpec.setCruiseControlSpec(cruiseControl);
```

Reply:
Ah my bad, you are talking about creating a CruiseControl object with the default fields! I had the same problem with this when writing the unit tests and should have made this code more testable back then. Anyways, to get around this, can you do something like:
```
CruiseControlConfiguration configuration = new CruiseControlConfiguration(new HashMap<String, Object>() {{
        putAll(configuration.getCruiseControlDefaultPropertiesMap());
    }}.entrySet()
    );
CruiseControlSpec cruiseControlSpec = new CruiseControlSpec()
cruiseControlSpec.setConfig((Map) configuration.asOrderedProperties().asMap());
```

Reply:
@kyguy it is not accesible. I have to add cluster-operator module but that does not make any sense. I have changed it like this to be more readable
```
CruiseControlSpec cruiseControl = new CruiseControlSpecBuilder()
                .withNewBrokerCapacity()
                    .withNewDisk("2000M")
                .endBrokerCapacity()
                .build();
            kafka.getSpec().setCruiseControl(cruiseControl);
```
and it works...